plugins/drng/plugin.go

```diff
@@ -68,6 +69,9 @@ func configureEvents() {
 		cachedMessageMetadata.Release()
 
 		cachedMessage.Consume(func(msg *message.Message) {
+			if len(msg.Payload().Bytes()) < header.Length {
```

Comment:
We should probably come up with a standard template how to handle messages in a solidification event of the base tangle. Does the following make sense?

- check `msg.Payload().Type()`. This is not a guarantee since anybody could send a message with your type but already a good and fast filter
- the payload interface should provide a `Size` method. Since this information is given in the message and can be verified this is also a quite fast check. Correct? @hmoog 
- then do any additional application specific checks

Reply:
Good points. I've added the Type() check. 
For the Size, I'm not sure. Should we trust whatever written in the size field of the message? Using len(payload.Bytes()) should be more safe. Wdyt?

Reply:
I think the base layer could verify it and store it. Not sure if it actually does this currently. That's why I linked Hans. What happens when the payload is shorter than the bytes in its header?

Reply:
Nothing critical, just the error you noticed: 
`tried to read 5 bytes from 4 bytes input`
that happens while parsing the message

Reply:
I usually simply "Parse" the payload and try to transform it into an object of the corresponding type. If the parsing fails, then we know its an invalid message and it gets dropped.

Reply:
Yeah I do the same (line 76). Should I then just remove the subsequent log?

Reply:
Maybe a debug if it's useful? But I don't really mind since it shouldn't appear anymore on every message anyway.